ext/web/13_message_port.js

```diff
@@ -239,30 +236,25 @@
     return [data, transferables];
   }
 
-  const detachedArrayBuffers = new WeakSet();
-
   /**
    * @param {any} data
    * @param {object[]} transferables
    * @returns {globalThis.__bootstrap.messagePort.MessageData}
    */
   function serializeJsMessageData(data, transferables) {
-    const transferredArrayBuffers = ArrayPrototypeFilter(
-      transferables,
-      (a) => ObjectPrototypeIsPrototypeOf(ArrayBufferPrototype, a),
-    );
-
-    for (const arrayBuffer of transferredArrayBuffers) {
-      // This is hacky with both false positives and false negatives for
-      // detecting detached array buffers. V8  needs to add a way to tell if a
-      // buffer is detached or not.
-      if (WeakSetPrototypeHas(detachedArrayBuffers, arrayBuffer)) {
-        throw new DOMException(
-          "Can not transfer detached ArrayBuffer",
-          "DataCloneError",
-        );
+    const transferredArrayBuffers = [];
+    for (let i = 0, j = 0; i < transferables.length; i++) {
+      const ab = transferables[i];
+      if (ObjectPrototypeIsPrototypeOf(ArrayBufferPrototype, ab)) {
+        if (ab.byteLength === 0 && core.ops.op_arraybuffer_was_detached(ab)) {
+          throw new DOMException(
+            `ArrayBuffer at index ${j} is already detached`,
+            "DataCloneError",
+          );
+        }
+        j++;
+        transferredArrayBuffers.push(ab);
```

Comment:
```suggestion
    for (let i = 0; i < transferables.length; i++) {
      const ab = transferables[i];
      if (ObjectPrototypeIsPrototypeOf(ArrayBufferPrototype, ab)) {
        if (ab.byteLength === 0 && core.ops.op_arraybuffer_was_detached(ab)) {
          throw new DOMException(
            `ArrayBuffer at index ${i} is already detached`,
            "DataCloneError",
          );
        }
        transferredArrayBuffers.push(ab);
```

Reply:
@lucacasonato Wanted to mimic Chrome error message.

```
const buf = new Uint8Array(5);
structuredClone(buf, { transfer: [buf.buffer] });
structuredClone([buf.buffer], { transfer: [new ReadableStream(), buf.buffer] });
// Uncaught DOMException: ... ArrayBuffer at index 0 is already detached
```

Also did it that way to make JS error be the same as Rust's where we're looping through AB list, not `transferables`

https://github.com/denoland/deno/blob/d3611550622749ab1d0fa606d31514090ac57998/core/ops_builtin_v8.rs#L471

Since we're not passing the `transferables` array to Rust, should I just remove the index hint, and leave the old message: `Can not transfer detached ArrayBuffer`?

Reply:
Okay, I think mimicing Chrome is fine here.